buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1116,7 +1116,10 @@ class BeamModulePlugin implements Plugin<Project> {
               }
               artifact project.sourcesJar
               artifact project.testSourcesJar
-              artifact project.javadocJar
+
+              if (!project.hasProperty("skipJavadoc")) {
```

Comment:
You should really add documentation here as to that this is used internally in Google otherwise the context as to why this flag was added will be hard to find.